src/service-providers/decision-trees/decision-trees.provider.ts

```diff
@@ -0,0 +1,106 @@
+import { AxiosRequestConfig } from 'axios'
+
+import { ServiceProviderBase } from '../common'
+import { filteringProvider } from '../filtering-regime'
+import { adoptDtreeStatResponse } from './decision-trees.adapters'
+import {
+  IDtreeStatArguments,
+  IDtreeStatResponse,
+} from './decision-trees.interface'
+import { TDtreeStat } from './decision-trees.types'
+import { getIncompleteProps } from './decision-trees.utils'
+
+type TGetFullDtreeStatParams = {
+  ds: string
+  no: string
+  code: string
+}
+
+type TGetFullDtreeStatOptions = {
+  abortSignal?: AbortSignal
+  onPartialResponse?: (response: TDtreeStat) => void
+}
```

Comment:
move types to decision-trees.interface.ts

Reply:
done